hack/update-codegen.sh

```diff
@@ -762,6 +762,43 @@ function codegen::subprojects() {
     done
 }
 
+function codegen::protobindings() {
+    # Each element of this array is a directory containing subdirectories which
+    # eventually contain a file named "api.proto".
+    local apis=(
+        "staging/src/k8s.io/cri-api/pkg/apis/runtime"
+
+        "staging/src/k8s.io/kubelet/pkg/apis/podresources"
+
+        "staging/src/k8s.io/kubelet/pkg/apis/deviceplugin"
+
+        "staging/src/k8s.io/kms/apis"
+        "staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2"
+
+        "staging/src/k8s.io/kubelet/pkg/apis/dra"
+
+        "staging/src/k8s.io/kubelet/pkg/apis/pluginregistration"
+        "pkg/kubelet/pluginmanager/pluginwatcher/example_plugin_apis"
+    )
+
+    kube::log::status "Generating protobuf bindings for ${#apis[@]} targets"
+    if [[ "${DBG_CODEGEN}" == 1 ]]; then
+        kube::log::status "DBG: generating protobuf bindings for:"
+        for dir in "${apis[@]}"; do
+            kube::log::status "DBG:     $dir"
+        done
+    fi
+
+    for api in "${apis[@]}"; do
+        git ls-files -z -cmo --exclude-standard ":(glob)${api}"/'**/api.pb.go' \
+            | xargs -0 rm -f
+    done
+
+    # NOTE: All output from this script needs to be copied back to the calling
+    # source tree.  This is managed in kube::build::copy_output in build/common.sh.
+    # If the output set is changed update that function.
+    build/run.sh hack/update-generated-proto-bindings-dockerized.sh "${apis[@]}"
```

Comment:
We can go ahead with this and I can take on a follow-up to run the proto generation uncontainerized if the right version of protoc is available in the host env.

As someone without a docker runtime on their primary dev machine, adding a `build/run.sh` invocation to update-codegen.sh makes me sad. I'm used to having to switch machines to run the full `make update` and proto generation scripts, but was hoping the docker requirement could remain limited to that piece (also, running uncontainerized is way faster, regardless of platform)

Reply:
SGTM, though IMO the whole build should be in docker and ~hermetic.  Local toolchains are a time bomb